exp/api/v1beta1/azureasomanagedcluster_types.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
```

Comment:
Should we introduce this as v1alpha1 and bake in alpha for at least one CAPZ release cycle?

Reply:
Yeah I think we could do that. I might've thrown myself off a little bit thinking that the metadata.yaml required us to have a single API version for all resources, but I don't think that's the case.